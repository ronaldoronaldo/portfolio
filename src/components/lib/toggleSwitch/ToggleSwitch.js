import React from 'react'
import { SliderContainer, Slider } from './ToggleSwitch.style'
import PropTypes from 'prop-types'

const ToggleSwitch = ({ onChange, active, disabled, ...rest }) => {
  return (
    <SliderContainer active={active} disabled={disabled} onClick={onChange} {...rest} >
      <Slider active={active} disabled={disabled} />
    </SliderContainer>
  )
}

ToggleSwitch.propTypes = {
  onChange: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
  disabled: PropTypes.bool,
}

export default ToggleSwitch
