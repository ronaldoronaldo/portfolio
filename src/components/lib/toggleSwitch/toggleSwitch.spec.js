import React from 'react'
import { shallow } from 'enzyme'
import ToggleSwitch from './ToggleSwitch'

describe('<ToggleSwitch>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <ToggleSwitch active="titulo" onChange={() => console.log('close')} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
