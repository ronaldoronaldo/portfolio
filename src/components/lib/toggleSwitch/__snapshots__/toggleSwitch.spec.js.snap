// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch> should render correctly 1`] = `ShallowWrapper {}`;
